### DISCUSSION TITLE
The safeTensors file generated by Lora cannot be used in webui

### DISCUSSION THREAD

**Discussion by kkiqq22** [cloneofsimo/lora#156] [Q&A]
The environment is the latest version of webui and cloneofsimo/ora
error：
activating extra network lora with arguments [<modules.extra_networks.ExtraNetworkParams object at 0x7fb832ac31f0>]: ValueError
Traceback (most recent call last):
File "/home/uqwenf/stable-diffusion-webui/modules/extra_networks.py", line 75, in activate
extra_network.activate(p, extra_network_args)
File "/home/uqwenf/stable-diffusion-webui/extensions-builtin/Lora/extra_networks_lora.py", line 23, in activate
lora.load_loras(names, multipliers)
File "/home/uqwenf/stable-diffusion-webui/extensions-builtin/Lora/lora.py", line 151, in load_loras
lora = load_lora(name, lora_on_disk.filename)
File "/home/uqwenf/stable-diffusion-webui/extensions-builtin/Lora/lora.py", line 89, in load_lora
key, lora_key = fullkey.split(".", 1)
ValueError: not enough values to unpack (expected 2, got 1)

Steps to reproduce the problem
Select model anything v4.5, Press show extra network.
Select LORA tab, click refresh
Press my LORA file, a prompt lora:A45CGKOTOKA_300:1 shows in prompts section.
(Prompt:lora:A45CGKOTOKA_300:1, masterpiece, best quality,1girl)
Image can be generated but only showing generic waifu, removing lora:A45CGKOTOKA_300:1 shows generic anime waifu

What should have happened?
Generic my target waifu as in "generate sample images" function in dreambooth extension.

**Comment by kkiqq22** [comment]
The size of the model file trained by some LORA frameworks is 288mb. Why is the LORA file trained by cloneofsimo/lora only 2-6mb?

**Reply by brian6091** [reply]
The size will depend on where LoRA was applied (e.g., only some matrices of the attention heads or  the feedfoward nets as well), and the rank used in each of these applications. This repo defaults to lower rank (I think 4?), and I've seen some others at r=128, so you will see a large size variation.

**Reply by kkiqq22** [reply]
Thanks for your reply, I will try lora_rank=4-128.
May I ask if you have checked this person's project, the lora file webui generated by his project can work
https://github.com/kohya-ss/sd-scripts

**Reply by DarkAlchy** [reply]
> Thanks for your reply, I will try lora_rank=4-128. May I ask if you have checked this person's project, the lora file webui generated by his project can work https://github.com/kohya-ss/sd-scripts

Yes, a fantastic little program (the webgui based on his _ss) but he will not port it to colab as he despises colab.  The results are absolutely amazing my friend is getting on his 3090.  The other dev who made a colab version is rubbish that wasted my time for far too long.  I don't know of anyone getting good results with it, but that webui version just flat out works.

**Comment by yankeguo** [comment]
I encountered same issue, I'm still investigating, seems `lora_pti` generates different key names

https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/a9fed7c364061ae6efb37f797b6b522cb3cf7aa2/extensions-builtin/Lora/lora.py#L17

**Reply by yankeguo** [reply]
keys of LoRA usable in `sdwebui`:

```
key_diffusers: lora_te_text_model_encoder_layers_0_mlp_fc1.alpha
key_diffusers: lora_te_text_model_encoder_layers_0_mlp_fc1.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_0_mlp_fc1.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_0_mlp_fc2.alpha
key_diffusers: lora_te_text_model_encoder_layers_0_mlp_fc2.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_0_mlp_fc2.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_k_proj.alpha
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_k_proj.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_k_proj.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_out_proj.alpha
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_out_proj.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_out_proj.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_q_proj.alpha
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_q_proj.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_q_proj.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_v_proj.alpha
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_v_proj.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_0_self_attn_v_proj.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_10_mlp_fc1.alpha
key_diffusers: lora_te_text_model_encoder_layers_10_mlp_fc1.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_10_mlp_fc1.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_10_mlp_fc2.alpha
key_diffusers: lora_te_text_model_encoder_layers_10_mlp_fc2.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_10_mlp_fc2.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_k_proj.alpha
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_k_proj.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_k_proj.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_out_proj.alpha
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_out_proj.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_out_proj.lora_up.weight
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_q_proj.alpha
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_q_proj.lora_down.weight
key_diffusers: lora_te_text_model_encoder_layers_10_self_attn_q_proj.lora_up.weight
....
```
keys of `lora_pti` generated model:

```
key_diffusers: text_encoder:5:down
key_diffusers: text_encoder:5:up
key_diffusers: text_encoder:6:down
key_diffusers: text_encoder:6:up
key_diffusers: text_encoder:7:down
key_diffusers: text_encoder:7:up
key_diffusers: text_encoder:8:down
key_diffusers: text_encoder:8:up
key_diffusers: text_encoder:9:down
key_diffusers: text_encoder:9:up
key_diffusers: unet:0:down
key_diffusers: unet:0:up
key_diffusers: unet:100:down
key_diffusers: unet:100:up
key_diffusers: unet:101:down
key_diffusers: unet:101:up
key_diffusers: unet:102:down
key_diffusers: unet:102:up
key_diffusers: unet:103:down
key_diffusers: unet:103:up
key_diffusers: unet:104:down
key_diffusers: unet:104:up
key_diffusers: unet:105:down
key_diffusers: unet:105:up
key_diffusers: unet:106:down
key_diffusers: unet:106:up
...
```

I hope we could find a way to make them compatible
